src/netCDF4/__init__.pyi

```diff
@@ -26,48 +86,119 @@ if sys.version_info >= (3, 10):
 elif not TYPE_CHECKING:
     ellipsis = type(Ellipsis)  # keeps ruff happy until ruff uses typeshed
 
-
-_DatatypeStrOptions: TypeAlias = Literal[
-    'S1', 'c', 'i1', 'b', 'B', 'u1', 'i2', 'h', 's', 'u2', 'i4',
-    'i', 'l', 'u4', 'i8', 'u8', 'f4', 'f', 'f8', 'd', 'c8', 'c16'
+# string type specifiers
+# fmt: off
+RealTypeLiteral: TypeAlias = Literal[
+    "i1", "b", "B", "int8",        # NC_BYTE
+    "u1", "uint8",                 # NC_UBYTE
+    "i2", "h", "s", "int16",       # NC_SHORT
+    "u2", "uint16",                # NC_USHORT
+    "i4", "i", "l", "int32",       # NC_INT
+    "u4", "uint32",                # NC_UINT
+    "i8", "int64", "int",          # NC_INT64
+    "u8", "uint64",                # NC_UINT64
+    "f4", "f", "float32",          # NC_FLOAT
+    "f8", "d", "float64", "float"  # NC_DOUBLE
 ]
-_DatatypeNCOptions: TypeAlias = Union[CompoundType, VLType, EnumType]
-DatatypeOptions: TypeAlias = Union[_DatatypeStrOptions, _DatatypeNCOptions, npt.DTypeLike]
-T_Datatype = TypeVar("T_Datatype", bound=DatatypeOptions)
-T_DatatypeNC = TypeVar("T_DatatypeNC", CompoundType, VLType, EnumType)
-
-DimensionsOptions: TypeAlias = Union[str, bytes, Dimension, Iterable[Union[str, bytes, Dimension]]]
-CompressionOptions: TypeAlias = Literal[
-    'zlib', 'szip', 'zstd', 'blosc_lz','blosc_lz4', 
-    'blosc_lz4hc', 'blosc_zlib', 'blosc_zstd'
-]
-CompressionLevelOptions: TypeAlias = Literal[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
-AccessModeOptions: TypeAlias = Literal['r', 'w', 'r+', 'a', 'x', 'rs', 'ws', 'r+s', 'as']
-FormatOptions: TypeAlias = Literal[
-    'NETCDF4', 'NETCDF4_CLASSIC', 'NETCDF3_CLASSIC',
-    'NETCDF3_64BIT_OFFSET', 'NETCDF3_64BIT_DATA'
+# fmt: on
+ComplexTypeLiteral: TypeAlias = Literal["c8", "c16", "complex64", "complex128"]
```

Comment:
I understand what you're saying. However there needs to be a distinction in the name that this is a Literal, i.e. a `str` with a fully-defined set of possible values, rather than a `type`. I'm open to other suggestions but I think `ComplexType` implies that a value of this type is a `type`, e.g. `np.complex32`.